src/main/java/junitparams/internal/ParameterisedTestMethodRunner.java

```diff
@@ -63,7 +63,7 @@ private Description findChildForParams(Statement methodInvoker, Description meth
         for (Description child : methodDescription.getChildren()) {
             InvokeParameterisedMethod parameterisedInvoker = findParameterisedMethodInvokerInChain(methodInvoker);
 
-            if (child.getMethodName().startsWith(parameterisedInvoker.getParamsAsString()))
+            if (child.hashCode() == parameterisedInvoker.getUniqueMethodId().hashCode())
```

Comment:
Hi, Michael!

I know, it smells very bad, but I wasn't able to find a clean solution here. As I understood, ParameterisedTestMethodRunner tries to find a corresponding child description by method invoker (which, basically, holds a concrete parameters set). The previous version used a method name for this mapping, assuming that we can reconstruct this name by the given parameters set. Now, the test case name is constructed by the special strategy, so we can't reconstruct this name, without explicit usage of this strategy. 

In order to solve it, I noticed the uniqueId parameter of junit Description class. According to the junit sources, it is used in the Description hashCode method, so I used the previous name (which had a pattern like [{index}] {method} ({params})) as a unique parameters set identifier, reconstructing it from parameters in InvokeParameterisedMethod class.

Anyway, I'll try to rethink this issue, and I'll be happy to hear any ideas about this problem.

P.S: Thank you for review! I resolved some comments, but I noticed few typos is TestCaseName.value() javadoc, going to fix it in the next commit.